docs/cheatsheet.rst

```diff
@@ -168,3 +168,42 @@ Modifiers
+:ref:`d <var-decl-initializer>`
+:ref:`E <equal-store-eliminator>`           Disambiguator, ForLoopInitRewriter                    ExpressionSplitter, SSATransform, CommonSubexpressionEliminator
+:ref:`e <expression-inliner>`               Disambiguator
+:ref:`F <function-specializer>`             Disambiguator, FunctionHoister                        LiteralRematerialiser
+:ref:`f <block-flattener>`                  Disambiguator, FunctionGrouper
+:ref:`g <function-grouper>`                 Disambiguator, FunctionHoister
+:ref:`h <function-hoister>`                 Disambiguator
+:ref:`I <for-loop-condition-into-body>`     Disambiguator                                         StructuralSimplifier
+:ref:`i <full-inliner>`                     Disambiguator                                         FunctionHoister, ExpressionSplitter
+:ref:`j <expression-joiner>`                Disambiguator
+:ref:`L <load-resolver>`                    Disambiguator, ForLoopInitRewriter                    SSATransform
+:ref:`l <circular-references-pruner>`       Disambiguator, FunctionHoister
+:ref:`M <loop-invariant-code-motion>`       Disambiguator, ForLoopInitRewriter, FunctionHoister   ExpressionSplitter, SSATransform, ForLoopConditionIntoBody
+:ref:`m <rematerialiser>`                   Disambiguator, ForLoopInitRewriter
+:ref:`n <control-flow-simplifier>`          Disambiguator, ForLoopInitRewriter, FunctionHoister
+:ref:`O <for-loop-condition-out-of-body>`                                                         LiteralRematerialiser
+:ref:`o <for-loop-init-rewriter>`           Disambiguator
+:ref:`p <unused-function-parameter-pruner>` Disambiguator, FunctionHoister                        LiteralRematerialiser                                           ExpressionInliner, UnusedPruner
+:ref:`r <unused-assign-eliminator>`         Disambiguator, ForLoopInitRewriter
+:ref:`S <unused-store-eliminator>`          Disambiguator, ForLoopInitRewriter                    SSATransform
+:ref:`s <expression-simplifier>`            Disambiguator, ForLoopInitRewriter                    ExpressionSplitter, SSATransform, CommonSubexpressionEliminator
+:ref:`T <literal-rematerialiser>`           Disambiguator, ForLoopInitRewriter
+:ref:`t <structural-simplifier>`            Disambiguator                                         LiteralRematerialiser
+:ref:`U <conditional-unsimplifier>`
+:ref:`u <unused-pruner>`                    Disambiguator
+:ref:`V <ssa-reverser>`                     Disambiguator                                                                                                         CommonSubexpressionEliminator, UnusedPruner
+:ref:`v <equivalent-function-combiner>`     Disambiguator, FunctionHoister                                                                                        UnusedPruner
+:ref:`x <expression-splitter>`                                                                    ForLoopConditionIntoBody
```

Comment:
While it may be true, it's a stretch to assume that this *doesn't* depend on the `Disambiguator` - the idea is that *all* steps do and there's no value on trying to be more precise than that. The same goes for the block flattener and the function hoister and the for loop init rewriter  - basically the ``hgfo`` sequence we have due to that hardcoded and non-configurable in the sequence.
There's also not much point in omitting them for some steps as hard dependency, since disabling them is not a configuration option anyways :-).

Reply:
And: slightly overapproximating is fully harmless here - while understating may lead to damage :-).

Reply:
I removed info about prerequisites from the optimizer docs and from the table. Now they're only in the docstrings. I also added a paragraph at the beginning of "Preprocessing" listing prerequisite steps and saying that all other steps implicitly depend on them.